ci/integration_tests.sh

```diff
@@ -11,6 +11,10 @@ export SPEC_OPTS="--order rand --format documentation"
 export CI=true
 export OSS=true
 
+if [ -n "$BUILD_JAVA_HOME" ]; then
```

Comment:
I don't think this approach works - at least not for integration tests as they currently stand. Note how [logstash is spawned](https://github.com/elastic/logstash/blob/master/qa/integration/services/logstash_service.rb#L134) in integration tests, using the ChildProcess gem calling `bin/logstash`. This will use the JDK pointed to by `JAVA_HOME` rather than the one spawned by the gradle process, meaning that integration tests will still be running Logstash using the installed JDK rather than any custom one specified in an environment variable.

(We can test how these tests would work on Jenkins by exporting a `$JDK` env value of one of the JDK's that should be already installed on Jenkins build machines - see https://github.com/elastic/infra/blob/7dcfe2930f82a1766bb6c52b03536786e7a5da82/ansible/roles/java/defaults/main.yml for details of locations)

Reply:
Great point, I haven't noticed, now I should have fixed it, from the point where the spec spawn the Logstash process, it sets JAVA_HOME of the forked process to the one read from system properties.

Reply:
Cool - can we now try a test run on Jenkins setting $JDK to verify that the correct JDK is being picked up by the tests?

Reply:
yes, I'll try one run changing the config in one of the LS projects in Jenkins, WDYT?

Reply:
Yes please